cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/network/model/v7/ConsentStatusApiModel.kt

```diff
@@ -0,0 +1,95 @@
+package com.sourcepoint.cmplibrary.data.network.model.v7
+
+import com.sourcepoint.cmplibrary.data.network.util.Env
+import com.sourcepoint.cmplibrary.model.exposed.CcpaStatus
+import com.sourcepoint.cmplibrary.model.exposed.GDPRPurposeGrants
+import org.json.JSONObject
+
+internal class ConsentStatusResp(
+    val thisContent: JSONObject,
+    val consentStatusData: ConsentStatusData,
+    val localState: JSONObject
+)
+
+internal data class ConsentStatusParamReq(
+    val env: Env,
+    val metadata: String,
+    val propertyId: Int,
+    val accountId: Int,
+    val authId: String? = null
+)
+
+internal data class ConsentStatusData(
+    val thisContent: JSONObject,
+    val gdprCS: GdprCS?,
+    val ccpaCS: CcpaCS?
+)
+
+internal data class GdprCS(
+    val thisContent: JSONObject,
+    var grants: Map<String, GDPRPurposeGrants> = emptyMap(),
+    val euconsent: String,
+    val addtlConsent: String,
+    val dateCreated: String,
+    val consentUUID: String,
+    val vendorListId: String,
+    val uuid: String,
+    val gdprApplies: Boolean,
+    val localDataCurrent: Boolean,
+    val cookieExpirationDays: Int,
+    val consentStatus: ConsentStatusCS
+)
+
+internal data class CustomVendorsResponse(
+    val consentedVendors: List<String>,
+    val consentedPurposes: List<String>,
+    val legIntPurposes: List<String>,
+)
+
+internal data class CcpaCS(
+    val thisContent: JSONObject,
+    val dateCreated: String?,
```

Comment:
Should `dateCreated` be a `Date` of some sort? On iOS I created a class called `SPDateCreated` just so I could encapsulate the parsing of the date format.

Reply:
I can create an extension on top of this class, 
for serialising data in Java, strings are the best option